### PR TITLE
MPSL: Always enable calibration when RC is used

### PR DESCRIPTION
**Pull request by rugeGerritsen**
MPSL requires RC clock calibration to be enabled when the RC clock is used as the Low Frequency clock source.

Ref: DRGN-15979

### CODE REVIEW

File: subsys/mpsl/mpsl_init.c [hunk 1]
```diff
@@ -155,7 +155,10 @@ static int mpsl_lib_init(const struct device *dev)
 	clock_cfg.skip_wait_lfclk_started =
 		IS_ENABLED(CONFIG_SYSTEM_CLOCK_NO_WAIT);
 
-#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION
+#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC
+#ifndef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION
+#error "MPSL requires clock calibration to be enabled when RC is used as LFCLK"
```
Comment (joerchan) [inline subsys/mpsl/mpsl_init.c:160]: Couldn't this be expressed in Kconfig? If not a BUILD_ASSERT would be better I think.

Comment (rugeGerritsen) [reply subsys/mpsl/mpsl_init.c:160]: I'm not sure how to express it in Kconfig in a nice way. `CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION` has a dependency to `CLOCK_CONTROL_NRF_K32SRC_RC`. Therefore we can't make `CLOCK_CONTROL_NRF_K32SRC_RC` select `CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION`. As `CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION` is a bool, we can't change its range. 

I can change it to use a build assert if you prefer that.

Comment (joerchan) [reply subsys/mpsl/mpsl_init.c:160]: I would have thought this:
```
	# MPSL requires clock calibration to be enabled when RC is used as LFCLK
	select CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION if CLOCK_CONTROL_NRF_K32SRC_RC
```
But looks like it doesn't work exactly how I expected it too.

Let's change it to a BUILD_ASSERT then.

Comment (rugeGerritsen) [reply subsys/mpsl/mpsl_init.c:160]: Tried using `BUILD_ASSERT(CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION, ""`. That does not work as `CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION` is not defined when set to `n`.

File: subsys/mpsl/mpsl_init.c [hunk 2]
```diff
@@ -155,7 +155,10 @@ static int mpsl_lib_init(const struct device *dev)
...
+#endif
```
Comment (joerchan) [inline subsys/mpsl/mpsl_init.c:161]:
```suggestion
#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC
BUILD_ASSERT(CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION,
 "MPSL requires clock calibration to be enabled when RC is used as LFCLK");
#endif
```

Comment (rugeGerritsen) [reply subsys/mpsl/mpsl_init.c]: Used `BUILD_ASSERT(IS_ENABLED(` as CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION may not be defined

**Review by wiba-nordic — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Review by carlopaparo — APPROVED** [review]

**Comment by joerchan** [comment]
@rugeGerritsen doc-required. Need a doc-person approval.

**Comment by joerchan** [comment]
Ah, sorry, mixing up who's who.

**Pull request merged**